### PR TITLE
add support to scrape k8s metrics

### PR DESCRIPTION
**Pull request by defo89**
Prepare Kubernetes API components to be scraped with `ServiceMonitor` of Prometheus Operator in `kube-monitoring-virtual`.

Hence, metrics ports are exposed with additional services for metrics.

Additional labels were added because I didn't want to edit existing `app` labels.

### CODE REVIEW

File: charts/kube-master/templates/controller-manager.yaml [hunk 1]
```diff
@@ -82,6 +83,12 @@ spec:
               readOnly: true
       containers:
         - name: controller-manager
+{{- if semverCompare ">=1.12-0" .Capabilities.KubeVersion.GitVersion }}
```
Comment (databus23) [inline charts/kube-master/templates/controller-manager.yaml:86]: can we change this to `">= 1.12"`? Thats what we are using in this deployment in a different location?

File: charts/kube-master/templates/scheduler.yaml [hunk 2]
```diff
@@ -45,6 +46,12 @@ spec:
             name: {{ include "master.fullname" . }}
       containers:
         - name: scheduler
+{{- if semverCompare ">=1.12-0" .Capabilities.KubeVersion.GitVersion }}
```
Comment (databus23) [inline charts/kube-master/templates/scheduler.yaml:49]: same here I would suggest to change it to `">= 1.12"
`

File: charts/kube-master/templates/service-metrics.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  name: etcd-metrics
+spec:
+  ports:
+  - name: metrics
+    port: 8081
+    protocol: TCP
+    targetPort: 8081
+  selector:
+    component: etcd
+{{- end }}
```
Comment (databus23) [inline charts/kube-master/templates/service-metrics.yaml:46]: I think the `{{ end }}` should move up to only exclude the controller-manager and scheduler for <1.12 as we enable metrics for etcd in all cases.

File: charts/kube-master/charts/etcd/templates/deployment.yaml [hunk 4]
```diff
@@ -52,7 +52,8 @@ data:
       --listen-client-urls=https://0.0.0.0:2379
 {{- else }}
       --advertise-client-urls=http://${ETCD_IP}:2379 \
-      --listen-client-urls=http://0.0.0.0:2379
+      --listen-client-urls=http://0.0.0.0:2379 \
+      --listen-metrics-urls=http://0.0.0.0:8081
```
Comment (databus23) [inline charts/kube-master/charts/etcd/templates/deployment.yaml:56]: hmm we have secure enabled by default now. so I think if we leave it like this we don't get metrics for etcd, no? I think we should add the metrics flag in all cases (outside of the if statement). What do you think?

Comment (defo89) [reply charts/kube-master/charts/etcd/templates/deployment.yaml:57]: you are right, I have missed the `.Values.secure.enabled` bit

**Pull request merged**